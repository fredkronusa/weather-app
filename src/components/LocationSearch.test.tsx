import * as React from "react";

import LocationSearchComponent from "./LocationSearch";
import { render } from "@testing-library/react";

const mockSetLocation = jest.fn();

describe("<LocationSearchComponent />", () => {
  // SNAPSHOT
  it("should match snapshot", async () => {
    const { asFragment } = render(
      <LocationSearchComponent
        setLocation={() => mockSetLocation()}
        location="melbourne"
      />
    );

    expect(asFragment()).toMatchSnapshot();
  });

  it("should set the input field ", async () => {
    const { container } = render(
      <LocationSearchComponent
        setLocation={() => mockSetLocation()}
        location="melbourne"
      />
    );
    const citySearch = container.querySelector("input");
    expect(citySearch?.value).toBe("melbourne");
  });
});
