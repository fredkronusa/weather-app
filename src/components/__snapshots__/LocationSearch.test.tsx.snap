// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LocationSearchComponent /> should match snapshot 1`] = `
<DocumentFragment>
  <div>
    <label
      class="sc-AxjAm tnjaK"
      for="citySearch"
    >
      Search Weather
    </label>
    <input
      class="sc-AxirZ fnMjvo"
      id="citySearch"
      name="citySearch"
      type="text"
      value="melbourne"
    />
    <input
      class="sc-AxiKw vmVWB"
      type="button"
      value="GO"
    />
  </div>
</DocumentFragment>
`;
